### ISSUE TITLE
Possible compatibility issues with PHP 7

### ISSUE THREAD

**Issue by CodeHex** [bugsnag/bugsnag-wordpress#15]
This exception occurred when sending errors using the wordpress package using PHP7

```
Fatal error: Uncaught TypeError: Argument 3 passed to Bugsnag_Error::fromPHPException() must be an instance of Exception, instance of Error given, called in
/home/vagrant/Sites/example.com/public/app/plugins/bugsnag/bugsnag-php/Client.php on line 360 and defined in 
/home/vagrant/Sites/example.com/public/app/plugins/bugsnag/bugsnag-php/Error.php:31 

Stack trace: 
#0 /home/vagrant/Sites/example.com/public/app/plugins/bugsnag/bugsnag-php/Client.php(360): Bugsnag_Error::fromPHPException(Object(Bugsnag_Configuration), Object(Bugsnag_Diagnostics), Object(Error)) 
#1 [internal function]: Bugsnag_Client->exceptionHandler(Object(Error)) 
#2 {main} thrown in /home/vagrant/Sites/example.com/public/app/plugins/bugsnag/bugsnag-php/Error.php on line 31
```

**Comment by GrahamCampbell** [comment]
Yeh, we just need to download the newest v2 from the PHP library and put it in here.

**Comment by GrahamCampbell** [comment]
Wait, but actually, I already fixed this? https://github.com/bugsnag/bugsnag-wordpress/commit/b704bdb9be1df26c382def3618e7cd0c1d5ab93e

**Comment by GrahamCampbell** [comment]
Guess we just need to release 1.3.0?

**Comment by GrahamCampbell** [comment]
https://github.com/bugsnag/bugsnag-wordpress/pull/16

**Comment by GrahamCampbell** [comment]
Not sure what else we need to do regarding actually publishing this for use?

**Comment by mikebronner** [comment]
I will give it a go using the `dev-master` to see if it works (I originally reported the issue a few days back) and let you know.

**Comment by mikebronner** [comment]
I was able to reproduce the issue again:

```
Fatal error: Uncaught TypeError: Argument 3 passed to Bugsnag_Error::fromPHPException() must be an instance of Exception, instance of Error given, called in /home/vagrant/Sites/maureentaylor.com/public/app/plugins/bugsnag/bugsnag-php/Client.php on line 360 and defined in /home/vagrant/Sites/maureentaylor.com/public/app/plugins/bugsnag/bugsnag-php/Error.php:31 Stack trace: #0 /home/vagrant/Sites/maureentaylor.com/public/app/plugins/bugsnag/bugsnag-php/Client.php(360): Bugsnag_Error::fromPHPException(Object(Bugsnag_Configuration), Object(Bugsnag_Diagnostics), Object(Error)) #1 [internal function]: Bugsnag_Client->exceptionHandler(Object(Error)) #2 {main} thrown in /home/vagrant/Sites/maureentaylor.com/public/app/plugins/bugsnag/bugsnag-php/Error.php on line 31
```

Here's the code that caused it, where `$content` is a snippet of HTML:
```php
    $dom = new DOMDocument();
    $dom->loadHTML($content);

    $caption = $dom->documentElement();

    var_dump($caption); die();
```

**Comment by mikebronner** [comment]
This is probably a duplicate of #13 and #14.

**Comment by tillkruss** [comment]
I'm getting this error with v1.3.0:

```
Fatal error: Uncaught TypeError: Argument 3 passed to Bugsnag_Error::fromPHPException() must be an instance of Exception, instance of TypeError given, called in /app/web/app/plugins/bugsnag/bugsnag-php/Client.php on line 360 and defined in /app/web/app/plugins/bugsnag/bugsnag-php/Error.php:31
Stack trace:
#0 /app/web/app/plugins/bugsnag/bugsnag-php/Client.php(360): Bugsnag_Error::fromPHPException(Object(Bugsnag_Configuration), Object(Bugsnag_Diagnostics), Object(TypeError))
#1 [internal function]: Bugsnag_Client->exceptionHandler(Object(TypeError))
#2 {main}
  thrown in /app/web/app/plugins/bugsnag/bugsnag-php/Error.php on line 31

**Comment by kattrali** [comment]
Fixed via 1.2.0/1.3.0.
